File: src/com/l2kt/gameserver/network/clientpackets/RequestReplyStopPledgeWar.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.data.sql.ClanTable;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.SystemMessageId;

public final class RequestReplyStopPledgeWar extends L2GameClientPacket
{
	private int _answer;
	
	@Override
	protected void readImpl()
	{
		_answer = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		final Player requestor = activeChar.getActiveRequester();
		if (requestor == null)
			return;
		
		if (_answer == 1)
			ClanTable.getInstance().deleteClansWars(requestor.getClanId(), activeChar.getClanId());
		else
			requestor.sendPacket(SystemMessageId.REQUEST_TO_END_WAR_HAS_BEEN_DENIED);
		
		activeChar.setActiveRequester(null);
		requestor.onTransactionResponse();
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestReplyStopPledgeWar.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.sql.ClanTable
import com.l2kt.gameserver.network.SystemMessageId

class RequestReplyStopPledgeWar : L2GameClientPacket() {
    private var _answer: Int = 0

    override fun readImpl() {
        _answer = readD()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        val requestor = activeChar.activeRequester ?: return

        if (_answer == 1)
            ClanTable.deleteClansWars(requestor.clanId, activeChar.clanId)
        else
            requestor.sendPacket(SystemMessageId.REQUEST_TO_END_WAR_HAS_BEEN_DENIED)

        activeChar.activeRequester = null
        requestor.onTransactionResponse()
    }
}